panther_analysis_tool/rule.py

```diff
@@ -349,16 +468,16 @@ def _run_rule(self, event: PantherEvent) -> bool:
         # defaults to True and will pass the events thru
         if self.rule_type == TYPE_SCHEDULED_RULE and not hasattr(self._module, "rule"):
             return True
-        return self._run_command(self._module.rule, event, bool)
+        return self._run_command(self._module.rule, event, bool)  # type: ignore[attr-defined]
 
     def _get_alert_context(
-        self, event: PantherEvent, use_default_on_exception: bool = True
+        self, event: PantherEvent, function_defined: bool, use_default_on_exception: bool = True
     ) -> Optional[str]:
-        if not hasattr(self._module, "alert_context"):
```

Comment:
Note that we actually need to check the attribute existence only the first time, probably once the module is loaded (?). This is a rather small optimization both in terms of flow clarity and speed, so I left it for the future.